src/t-i18n.ts

```diff
@@ -67,14 +68,14 @@ export class I18n {
 			return defaultMessages[key];
 		}
 
-		return;
+		return "";
 	}
 
-	generateId(message: string) {
+	generateId(message: string): string {
 		return this.idGenerator(message);
 	}
 	
-	lookup(id: string, replacements: Replacements = null, defaultMessage:string = null):string {
+	lookup(id: string, replacements: Replacements = null, defaultMessage:string | null = null):string {
```

Comment:
These nulls just seem silly. The function takes two optional parameters. If they're left out they would be simply `undefined`. But here we're converting those to `null` instead. This is why we use `undefined` instead of `null` in the web client. The JS language prefers it and there's no reason to have both.

That said, TS's strict checks will catch these and enforce correct handling, so I'm not too concerned as a consumer of this library.

Reply:
The only reason I asked for `null` is to support this call signature:

```
T.lookup("some-id", null, "some default message");
```

It reads better and looks more intentional than passing `undefined` as a literal argument to a function. Maybe that's just something I have to get used to.

Reply:
I agree that the readability and implied intentionality of `null` is something that we're used to. And I get that. But I've become so tired of its idiosyncrasies.